### PR TITLE
New data source for security groups

### PR DESCRIPTION
**Pull request by malferov**

### CODE REVIEW

File: alicloud/data_source_alicloud_security_groups.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	"regexp"
+
+	"github.com/denverdino/aliyungo/ecs"
+	"github.com/denverdino/aliyungo/util"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+type SecurityGroup struct {
+	Attribute    ecs.DescribeSecurityGroupAttributeResponse
+	CreationTime util.ISO6801Time
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_security_groups.go]: I think type SecurityGroup is needless and ecs.DescribeSecurityGroupAttributeResponse is ok.

File: alicloud/data_source_alicloud_security_groups.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+	for {
+		items, paginationResult, err := conn.DescribeSecurityGroups(args)
+		if err != nil {
+			return fmt.Errorf("DescribeSecurityGroups: %#v", err)
+		}
+
+		for _, item := range items {
+			if v, ok := d.GetOk("vpc_id"); ok && v.(string) != item.VpcId {
+				continue
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_security_groups.go]: Line 100 may be needless and ecs.DescribeSecurityGroupsArgs support 'VpcId'.

File: alicloud/data_source_alicloud_security_groups.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+	if err := d.Set("groups", s); err != nil {
+		return err
+	}
+
+	// create a json file in current directory and write data source to it
+	if output, ok := d.GetOk("output_file"); ok && output.(string) != "" {
+		writeToFile(output.(string), s)
+	}
+	return nil
+}
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_security_groups.go]: Please add security groups' test cases.

File: alicloud/data_source_alicloud_security_groups.go (hunk 2 above)
Comment (malferov) [reply alicloud/data_source_alicloud_security_groups.go]: fixed

File: alicloud/data_source_alicloud_security_groups.go (hunk 3 above)
Comment (malferov) [reply alicloud/data_source_alicloud_security_groups.go]: added

File: alicloud/data_source_alicloud_security_groups.go (hunk 1 above)
Comment (xiaozhu36) [reply alicloud/data_source_alicloud_security_groups.go]: Hi @malferov Do you think it need a new type 'SecurityGroup' to support the next function?

Comment (xiaozhu36) [reply alicloud/data_source_alicloud_security_groups.go]: In addition, please add a docs for security groups datasource in folder terraform-provider-alicloud/website/docs/d/ . Thanks a lot.

Comment (malferov) [reply alicloud/data_source_alicloud_security_groups.go]: New type is needless indeed. Please find update.

Comment (malferov) [reply alicloud/data_source_alicloud_security_groups.go]: Docs have been added.

File: alicloud/data_source_alicloud_security_groups.go [hunk 4]
```diff
@@ -10,11 +10,6 @@ import (
 	"github.com/hashicorp/terraform/helper/schema"
 )
 
-type SecurityGroup struct {
-	Attribute    ecs.DescribeSecurityGroupAttributeResponse
-	CreationTime util.ISO6801Time
-}
-
 func dataSourceAlicloudSecurityGroups() *schema.Resource {
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_security_groups.go]: HI, I am sorry that I give you a incorrect review. I think security group CreationTime can be got from ecs.DescribeSecurityGroupAttributeResponse, and new type SecurityGroup is needless. But it is my fault and CreationTime does not exist in DescribeSecurityGroupAttributeResponse. So may be you should return. I am sorry again.

File: website/docs/d/security_groups.html.markdown [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+## Attributes Reference
+
+A list of security groups will be exported and its every element contains the following attributes:
+
+* `id` - The ID of the security group.
+* `name` - The name of the security group.
+* `description` - The description of the security group.
+* `vpc_id` - The ID of the VPC.
+* `inner_access` - Whether to allow inner network access.
+* `creation_time` - Creation time of the security group.
```
Comment (xiaozhu36) [inline website/docs/d/security_groups.html.markdown]: Hi, you should add a record in https://github.com/terraform-providers/terraform-provider-alicloud/blob/master/website/alicloud.erb and to show the docs in website.

File: alicloud/data_source_alicloud_security_groups.go (hunk 4 above)
Comment (malferov) [reply alicloud/data_source_alicloud_security_groups.go]: Hi, no problem at all. Thank you for the review and for clarification. Please find my update.

File: website/docs/d/security_groups.html.markdown (hunk 5 above)
Comment (malferov) [reply website/docs/d/security_groups.html.markdown]: done

**Pull request merged**